### PR TITLE
fix: update current-cloud backend-config on resource removal

### PR DESCRIPTION
**Pull request by edwardfoyle**
Update #current-cloud-backend/backend-config.json on resource removal
Also remove autoVerifiedAttributes from auth immutable attribute list as it is mutable: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpool.html#cfn-cognito-userpool-autoverifiedattributes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/src/push-resources.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ async function run(context, resourceDefinition) {
     await postPushGraphQLCodegen(context);
     await amplifyServiceManager.postPushCheck(context);
 
-    if (resources.length > 0) {
+    if (resources.concat(resourcesToBeDeleted).length > 0) {
```
Comment (kaustavghosh06) [inline packages/amplify-provider-awscloudformation/src/push-resources.js:71]: aren't resources all the resources - add/update/delete?

File: packages/amplify-e2e-core/src/utils/projectMeta.ts [hunk 2]
```diff
@@ -41,6 +41,11 @@ function getBackendConfig(projectRoot: string) {
   return JSON.parse(fs.readFileSync(backendFConfigFilePath, 'utf8'));
 }
 
+function getCloudBackendConfig(projectRoot: string) {
```
Comment (kaustavghosh06) [inline packages/amplify-e2e-core/src/utils/projectMeta.ts:44]: Are we using this anywhere other than tests?

File: packages/amplify-cli/src/extensions/amplify-helpers/update-amplify-meta.ts [hunk 3]
```diff
@@ -137,10 +140,12 @@ export async function updateamplifyMetaAfterPush(resources) {
     // Skip hash calculation for imported resources
     if (resources[i].serviceType !== 'imported') {
       const sourceDir = path.normalize(path.join(pathManager.getBackendDirPath(), resources[i].category, resources[i].resourceName));
-      const hashDir = await getHashForResourceDir(sourceDir);
-
-      amplifyMeta[resources[i].category][resources[i].resourceName].lastPushDirHash = hashDir;
-      amplifyMeta[resources[i].category][resources[i].resourceName].lastPushTimeStamp = currentTimestamp;
+      // skip hashing deleted resources
+      if (fs.pathExistsSync(sourceDir)) {
+        const hashDir = await getHashForResourceDir(sourceDir);
```
Comment (kaustavghosh06) [inline packages/amplify-cli/src/extensions/amplify-helpers/update-amplify-meta.ts:145]: Without calculating the hash - did you check if the resource status still shows as "Deleted"?

File: packages/amplify-e2e-core/src/utils/projectMeta.ts (hunk 2 above)
Comment (edwardfoyle) [reply packages/amplify-e2e-core/src/utils/projectMeta.ts:44]: no, I added it for the e2e test

File: packages/amplify-provider-awscloudformation/src/push-resources.js (hunk 1 above)
Comment (edwardfoyle) [reply packages/amplify-provider-awscloudformation/src/push-resources.js:71]: no, if you look above resources is just a concatenation of add and update which was the root cause of the bug

File: packages/amplify-cli/src/extensions/amplify-helpers/update-amplify-meta.ts (hunk 3 above)
Comment (edwardfoyle) [reply packages/amplify-cli/src/extensions/amplify-helpers/update-amplify-meta.ts:145]: after pushing, the deleted resource doesn't show up at all (which is expected)
Also checked delete -> push -> new folder -> pull -> status and the deleted resource doesn't show up

**Review by kaustavghosh06 — APPROVED** [review]
LGTM.

**Pull request merged**